File: PlayTheOdds/Models/Wager.cs
using System;
using System.Collections.Generic;

namespace PlayTheOdds.Models
{
    public class Wager
    {
        public int Id { get; set; }

        public DateTime Date { get; set; }

        public Team Team1 { get; set; }

        public Team Team2 { get; set; }

        public string Name { get; set; }

        public Odd OddLeft { get; set; }

        public Odd OddRight { get; set; }

        public Dictionary<string, string> AdditionalData { get; set; }
    }

    public class Odd
    {
        public Team Team { get; set; }

        public int Value { get; set; }
    }
}

File: PlayTheOdds/wwwroot/js/models/wager.ts
import Team from "../models/team"
import { WagerStatus } from "../enums/status"

export class Wager {
    public id: number;
    public date: Date;
    public mode: string;
    public status: WagerStatus;
    public team1: WagerTeam;
    public team2: WagerTeam;

    constructor(id: number, date: Date, mode: string, team1: WagerTeam, team2: WagerTeam) {
    }
}

export class WagerTeam {
    public handicap: string | null;
    public odd: number;
    public team: Team;

    constructor(team: Team, odd: number) {
    }
}